### PR TITLE
[NETBEANS-5599] PHP 8.1 Support: Pure intersection types

### PR DESCRIPTION
**Pull request by junichi11**
https://issues.apache.org/jira/browse/NETBEANS-5599

RFC: https://wiki.php.net/rfc/pure-intersection-types

Parser&Lexer:

- Fix the `ASTPHP5Parser.cup` and `ASTPHP5Scanner.flex`
- Fix the `PHP81UnhandledError`
- Add the `IntersectionType` as a new AST node
- Add and fix the unit tests for the parser
- Fix the Navigator

GotoDeclaration, MarkOccurrences, Formatter:

- Fix GotoDeclaration, MarkOccurrences, and Formatter features
- Add unit tests

Screenshots

#### PHP 8.1
nb-php81-pure-intersection-types

#### PHP 8.0
nb-php81-pure-intersection-types-php80

## Note

Other features have not been fixed yet. (e.g. Code completion)

### CODE REVIEW

**Comment by junichi11** [comment]
@tmysik Could you please have a look at this when you have time?
If you don't have time, please let me know. I'll look at this myself once again, then merge it because the feature freeze date is Jan 17th. (https://lists.apache.org/thread/8wjjx94p1w0c7o4f07p19oq2phcj9nzd)

I haven't fixed other features(e.g. Code completion, Hints) of intersection type yet. I haven't implemented the other features of PHP 8.1 (e.g. enum).
However, I'm going to merge the `php81-support` branch into the master after this PR is merged. I'm going to implement/fix the other features in NB14.
Thanks.

**Comment by junichi11** [comment]
@tmysik Need not hurry :) I can merge this after check again today.

**Review by tmysik — APPROVED** [review]

**Comment by junichi11** [comment]
@tmysik Thank you, Tomas!

**Pull request merged**
